acceptance_modelisation/toolbox.py

```diff
@@ -82,3 +84,68 @@ def get_unique_wobble_pointings(observations: Observations, max_angular_separati
     for key, value in wobbles_dict.items():
         logger.info(f"{key}: [{value[0].round(2)}, {value[1].round(2)}]")
     return wobbles
+
+
+def get_time_mini_irf(observation: Observation, mini_irf_time_resolution: u.Quantity):
+    """
+    Compute the evaluation time for each mini irfs
+
+    Parameters
+    ----------
+    observation : gammapy.data.observations.Observation
+        The observation for which mini irf will be generated
+    mini_irf_time_resolution : u.Quantity
+        The targeted time resolution for the mini irfs
+
+    Returns
+    -------
+    evaluation_time : u.Quantity
+        An array of the evaluation time of each mini irf, correspond to the central time of the bin
+    observation_time : u.Quantity
+        The observation time associated with each mini irf
+    """
+
+    total_observation_duration = observation.tstop - observation.tstart
+    nb_bin = int(np.ceil((total_observation_duration / mini_irf_time_resolution).to_value(u.dimensionless_unscaled)))
+    bin_duration = total_observation_duration / nb_bin
+    evaluation_time = Time(np.zeros(nb_bin), format='unix')
+    observation_time = np.zeros(nb_bin) * mini_irf_time_resolution.unit
+    for i in range(nb_bin):
+        evaluation_time[i] = observation.tstart + (i + 0.5) * bin_duration
+        observation_time[i] = bin_duration
+
+    return evaluation_time, observation_time
+
+
+def generate_irf_from_mini_irf(data_cube: np.array, observation_time: u.Quantity):
+    """
+    Compute the final irf from the collection of mini irfs
+
+    Parameters
+    ----------
+    data_cube : numpy.array
+        The data cube containing all the mini irfs. Axis 0 should be the mini irfs axis
+    observation_time : u.Quantity
+        The observation time for each bin
+
+    Returns
+    -------
+    data_cube_final : u.Quantity
+        The data cube containing the final irf
+    """
+
+    # Compute the scale factor for each mini irf to get a weighted average
+    scale_factor_per_bin = (observation_time / np.sum(observation_time)).to_value(u.dimensionless_unscaled)
+
+    # Reshape the scale factor in order to be able to multiply it to the data cube
+    shape_data_cube = list(data_cube.shape)
+    for i in range(len(shape_data_cube)):
+        if i != 0:
+            shape_data_cube[i] = 1
+    shape_data_cube = tuple(shape_data_cube)
+    scale_factor_per_bin_reshaped = scale_factor_per_bin.reshape(shape_data_cube)
```

Comment:
Here you have a scale factor of dimension 1 and len N, multiplying data of shape (N, ?, ?, ...) 
It should be possible without reshaping.

Also if you have a single scaling factor it is even easier

Reply:
I've got an error during my testing, and I just added this reshape to be sure of what will be done. As the supplementary dimension are of order one, it should not involve any copy and thus slow down the code.